src/torchcodec/decoders/_core/ops.py

```diff
@@ -26,19 +31,37 @@ def load_torchcodec_extension():
     # correct .so file, so this for-loop succeeds on the first iteration.
 
     exceptions = []
+    pybind_ops_module_name = "_torchcodec_pybind_ops"
     for ffmpeg_major_version in (7, 6, 5, 4):
-        library_name = f"libtorchcodec{ffmpeg_major_version}"
+        decoder_library_name = f"libtorchcodec_decoder{ffmpeg_major_version}"
+        custom_ops_library_name = f"libtorchcodec_custom_ops{ffmpeg_major_version}"
+        pybind_ops_library_name = f"libtorchcodec_pybind_ops{ffmpeg_major_version}"
         try:
-            torch.ops.load_library(_get_extension_path(library_name))
+            torch.ops.load_library(_get_extension_path(decoder_library_name))
+            torch.ops.load_library(_get_extension_path(custom_ops_library_name))
+
+            pybind_ops_library_path = _get_extension_path(pybind_ops_library_name)
+            torch.ops.load_library(pybind_ops_library_path)
+            spec = importlib.util.spec_from_file_location(
+                pybind_ops_module_name,
+                pybind_ops_library_path,
+            )
+            if spec is None:
+                raise ImportError(
+                    f"Unable to load spec for module {pybind_ops_module_name} from path {pybind_ops_library_path}"
+                )
+
+            global _pybind_ops
+            _pybind_ops = importlib.util.module_from_spec(spec)
```

Comment:
This does not yet work on Mac. See: https://github.com/pytorch/torchcodec/actions/runs/13924707065/job/38966320796?pr=564

We're unable to load the spec from the file on line 45. When digging around, it seems like we should be using `ctypes.CDLL` to load a shared library, and that's what `torch.ops.load_library()` does under the hood. Hence why I kept that in for the pybind ops.

TorchAudio does both: https://github.com/pytorch/audio/blob/c670ad81fda266b6598aeeef434583eb98197ae8/src/torio/_extension/utils.py#L105

The common case for pybind11 is that you have:

1. A shared library named `foo.so`.
2. A module declared as `foo` in the `PYBIND11_MODULE` macro in C++.
3. `foo.so` lives in a place Python knows to look.

Then `import foo` should just work.

Reply:
Works now! I'll make comments in the appropriate places.